ios/FluentUI/Core/Theme/Tokens/TokenizedControl.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// Defines a control with customizable design tokens.
+public protocol TokenizedControl: View {
+    associatedtype TokenType: ControlTokens
+
+    /// Modifier function that updates the design tokens for a given control.
+    ///
+    /// - Parameter tokens: The custom tokens to apply to this control.
+    ///
+    /// - Returns: A version of this control with these custom tokens applied.
+    func customTokens(_ tokens: TokenType) -> Self
+}
+
+/// Internal extension to `TokenizedControl` that adds information about `configuration`.
+protocol TokenizedControlInternal: TokenizedControl {
+    associatedtype ConfigurationType: ControlConfiguration where ConfigurationType.TokenType == TokenType
+
+    var state: ConfigurationType { get }
```

Comment:
* Yep, it'll break things when this changes. I'll file a new task, but since I assume it'll come with this change I'm at least referring to the new protocols as `Configuration`. The property is still `state`.
* Yep, changing `state`=>`configuration` will be a big change, and won't be part of this PR (or indeed part of this branch at all, most likely).
* Good question. Our SwiftUI controls should be fine using Configuration, since there's no inheritance to worry about. UIKit will be trickier though, so we'll need to be careful about adding that name (though we've largely avoided that problem so far).